experimental/oak_baremetal_app_qemu/src/multiboot.rs

```diff
@@ -14,56 +14,140 @@
 // limitations under the License.
 //
 
-// Required as Bindgen-generated code doesn't follow idiomatic Rust style.
-#![allow(non_camel_case_types)]
-
 use bitflags::bitflags;
 use core::ffi::{c_char, CStr};
 use oak_baremetal_kernel::boot::{BootInfo, E820Entry, E820EntryType};
 
-include!(concat!(env!("OUT_DIR"), "/multiboot.rs"));
+/// Magic constant that will be in the `EAX` register when loaded by Multiboot.
+pub const BOOT_MAGIC: u64 = 0x2BADB002;
+
+bitflags! {
+    /// Flags specify data available in the MultibootInfo data structure.
+    /// See <https://www.gnu.org/software/grub/manual/multiboot/multiboot.html#Boot-information-format>
+    /// for more details.
+    pub struct MultibootInfoFlags: u32 {
+        /// is there basic lower/upper memory information?
+        const MEMORY = 0x00000001;
+
+        /// is there a boot device set?
+        const BOOTDEV = 0x00000002;
+
+        /// is the command-line defined?
+        const CMDLINE = 0x00000004;
+
+        /// are there modules to do something with?
+        const MODS = 0x00000008;
+
+        /// is there a symbol table loaded?
+        /// Mutually exclusive witht ELF_SHDR>.
+        const AOUT_SYMS = 0x00000010;
```

Comment:
These may read better as binary constants, or bit-shifts (assuming that's a const operator, which I am not sure), but up to you. In particular, the bit-shift version seems easier to eyeball against the spec in https://www.gnu.org/software/grub/manual/multiboot/multiboot.html#Boot-information-format .

Reply:
Those were straight from the C header file, but fair enough, yes, bit shifts make these nicer to read.